### ISSUE TITLE
`SetProcessAppUserModelId` may throw when called from `onEveryRun`

### ISSUE THREAD

**Issue by peppy** [clowd/Clowd.Squirrel#60]
We've had a [report](https://github.com/ppy/osu/discussions/17190) from a developer that they are unable to start our project when running without a local squirrel install. It seems like this call should either not throw, or the fact that it does (and instructions on how/where it is safe to call for) should be added.

Our current usage can be found [here](https://github.com/ppy/osu/blob/b286122413b21be910bf29c8349f3030d3e4b2f3/osu.Desktop/Program.cs#L126) for reference. The implementation roughly matches that documented in the [quick start](https://github.com/clowd/Clowd.Squirrel#quick-start-for-net-apps) instructions.

Example call stack:

```csharp
System.InvalidOperationException: This operation is only valid in an installed application (Update.exe was not found).
Check 'IsInstalledApp' or 'CurrentlyInstalledVersion()' before calling this function.
   at Squirrel.UpdateManager.getUpdateExe() in ./UpdateManager.cs:line 399
   at Squirrel.UpdateManager.getInstalledApplicationName() in ./UpdateManager.cs:line 369
   at Squirrel.UpdateManager.get_AppId() in ./UpdateManager.cs:line 23
   at Squirrel.UpdateManager.get_AppDirectory() in ./UpdateManager.cs:line 26
   at Squirrel.UpdateManager.SetProcessAppUserModelId() in ./UpdateManager.cs:line 193
   at osu.Desktop.Program.<>c.<setupSquirrel>b__3_2(SemanticVersion version, IAppTools tools, Boolean firstRun) in D:\Projects\OSU\osu\osu.Desktop\Program.cs:line 126
   at Squirrel.SquirrelAwareApp.HandleEvents(SquirrelHook onInitialInstall, SquirrelHook onAppUpdate, SquirrelHook onAppObsoleted, SquirrelHook onAppUninstall, SquirrelRunHook onEveryRun, String[] arguments) in ./SquirrelAwareApp.cs:line 141
   at osu.Desktop.Program.setupSquirrel() in D:\Projects\OSU\osu\osu.Desktop\Program.cs:line 116
   at osu.Desktop.Program.Main(String[] args) in D:\Projects\OSU\osu\osu.Desktop\Program.cs:line 31
```

**Comment by caesay** [comment]
Fix released [in 2.8.28-pre](https://github.com/clowd/Clowd.Squirrel/releases/tag/2.8.28-pre).

**Comment by peppy** [comment]
Thanks for the quick fix and release!
